lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -874,18 +874,22 @@ def get_service(self, name):
         ds = DockerService()
 
         task_template_data = raw_data['Spec']['TaskTemplate']
-        update_config_data = raw_data['Spec']['UpdateConfig']
+        # Try/Except in case service already exists without UpdateConfig
+        try:
+            update_config_data = raw_data['Spec']['UpdateConfig']
+            ds.update_delay = update_config_data['Delay']
+            ds.update_parallelism = update_config_data['Parallelism']
+            ds.update_failure_action = update_config_data['FailureAction']
+            ds.update_monitor = update_config_data['Monitor']
+            ds.update_max_failure_ratio = update_config_data['MaxFailureRatio']
+            ds.update_order = update_config_data['Order']
+        except Exception as e:
```

Comment:
As noted in the PR summary, the condition you're attempting to check against is `KeyError`. What's the motivation to cast the net wider and catch all exceptions and pass when one is caught?

I'd prefer this to narrow the scope of the exception catching just in case some other exception were to occur, it isn't hidden from the user unless there's a reason for this. Thanks!

Reply:
It looks like `KeyError` is the only meaningful thing that could happen in this code.

Reply:
@maxamillion Sorry for the confusion but its not actually a KeyError.  The missing Key is the problem, so I said that to provide clarity on the situation.   The error returned is not actually very verbose or specific at all.

Reply:
So I've got a bit of a conundrum on this one.  I just spent some time reverting the code to older states and trying to recreate the issue so I could demonstrate.  After many attempts I was unsuccessful as triggering the issue again and every service I create on docker swarm  by default has an `UpdateConfig` stanza.

It's quite possible the issue I ran into was from a docker service created on older versions of docker-py or some other condition that no longer exists in my environments.

That being said, I do not believe this is a detrimental piece of code to add and I do believe that its still covers an edge-case that may spring up in the future.

Not sure how to proceed.

Reply:
I think nicer than catching the exception would be to stick to `KeyError` or to use `.get()` on the dictionary and add a conditional for if UpdateConfig is undefined, instead of a generic Exception-catch.

Reply:
I'm actually going to close this issue because I can't reproduce it anymore.  I'm uncomfortable adding code for a condition that cannot be reached.  If anyone is able to reproduce or feels otherwise please feel free to re-open